core/src/main/java/com/amplifyframework/util/UserAgent.java

```diff
@@ -128,7 +122,7 @@ private static String sanitize(@Nullable String string) {
                 return "UNKNOWN";
             }
 
-            return string.replace(' ', '_');
+            return string;
```

Comment:
Do you know why this was here before and why we no longer need it?

Reply:
Perhaps a bit of a drive-by change. I'm not sure why we were normalizing the output this way. @jamesonwilliams?

I removed it as it isn't customary to replace spaces with underscores in [User-Agent header values](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent) generally.

Reply:
Mohit Srivastava & Nikhil Dabhade had requested that we use the same format as is used in the SDK.

However, my guess is that not a ton of thought had gone into designing the SDK one, previously. More just that it had set a formatting precedence.

Anyway, the SDK had this `_` character, among other weird choices (several of which John is correcting in this PR.)

If the `_` is weird, let's get rid of it. The best outcome is a well-formatted and usable `User-Agent` string.